python_tests/test_csf.py

```diff
@@ -21,34 +21,37 @@ def run_csf_test(keys, values):
 
 
 @pytest.mark.unit
-def test_csf():
+def test_csf_int():
     keys = [f"key{i}" for i in range(1000)]
     values = [i for i in range(1000)]
-    run_csf_test(keys, values)
+    run_csf_int_test(keys, values)
 
 
 @pytest.mark.unit
-def test_csf_byte_strings():
+def test_csf_byte_string_keys():
     keys = [f"key{i}".encode("utf-8") for i in range(1000)]
     values = [i for i in range(1000)]
-    run_csf_test(keys, values)
+    run_csf_int_test(keys, values)
 
 
 @pytest.mark.unit
 def test_csf_char_10():
     keys = [f"key{i}".encode("utf-8") for i in range(1000)]
-    values = [list(f"value{i}".ljust(10)[:10]) for i in range(1000)]
+    values = [str(f"value{i}".ljust(10)[:10]) for i in range(1000)]
 
-    csf = caramel.CSFChar10(keys, values)
+    list_to_str = lambda x: ''.join(x)
+    csf = caramel.CSFQueryWrapper(caramel.CSFChar10(keys, values), list_to_str)
 
     for key, value in zip(keys, values):
+        print(csf.query, csf.query(key))
         assert csf.query(key) == value
 
     filename = "temp.csf"
     csf.save(filename)
-    csf = caramel.CSFChar10.load(filename)
+    csf = caramel.CSFQueryWrapper(caramel.CSFChar10.load(filename), list_to_str)
```

Comment:
What's the behavior for this wrapper when calling "save" on it?

Reply:
`wrapper.save` passes through all of the underlying CSF's methods, including `.save`, so it saves the underlying CSF as whatever type (in this case `CSFChar10`). The only difference between wrapper behavior and standard behavior is that the wrapper calls a post-processing function on the query results.